drivers/modem/gsm_ppp.c

```diff
@@ -382,20 +388,32 @@ static void gsm_finalize_connection(struct gsm_modem *gsm)
 		return;
 	}
 
+attaching:
 	/* Don't initialize PPP until we're attached to packet service */
-	ret = modem_cmd_send_nolock(&gsm->context.iface,
-				    &gsm->context.cmd_handler,
-				    &check_attached_cmd, 1,
-				    "AT+CGATT?",
-				    &gsm->sem_response,
-				    GSM_CMD_SETUP_TIMEOUT);
+	ret = modem_cmd_send_nolock(
+		&gsm->context.iface, &gsm->context.cmd_handler,
+		&check_attached_cmd, 1, "AT+CGATT?", &gsm->sem_response,
+		GSM_CMD_SETUP_TIMEOUT);
 	if (ret < 0) {
+		/*
+		 * attach_retries not set        -> trigger N attach retries
+		 * attach_retries set            -> decrement and retry
+		 * attach_retries set, becomes 0 -> trigger full retry
+		 */
+		if (!gsm->attach_retries) {
+			gsm->attach_retries = CONFIG_MODEM_GSM_ATTACH_TIMEOUT;
```

Comment:
```
...attach_retries = ...ATTACH_TIMEOUT;
```
This seems wrong. Either there is a timeout in seconds or there is a retry limit.

Reply:
>Well it retries at 1 Hz so they coincide.

If someone will change `K_SECONDS(1)` to some other value, then above statement will be wrong. Maybe use something like:
```
#define GSM_ATTACH_RETRY_DELAY_MSEC 1000
```
then in the code
```
gsm->attach_retries = CONFIG_MODEM_GSM_ATTACH_TIMEOUT * MSEC_PER_SEC / GSM_ATTACH_RETRY_DELAY_MSEC;
```
and then
```
void)k_delayed_work_submit(&gsm->gsm_configure_work,
			K_MSEC(GSM_ATTACH_RETRY_DELAY_MSEC));
```

Reply:
yeah looks clean enough I suppose, how about now? I'll squash as soon as it's approved

Reply:
I like it now. Please squash and I will approve this PR later on, so that we don't merge accidentally. Thanks!

Reply:
ping